stdlib/src/collections/list.mojo

```diff
@@ -414,6 +414,49 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
             earlier_idx += 1
             later_idx -= 1
 
+    # TODO: Modify these to regular methods when issue 1876 is resolved
+    @staticmethod
+    fn index[C: ComparableCollectionElement](self: List[C], owned value: C, start: Int = 0) raises -> Int:
```

Comment:
I don't think you need to implement the whole logic twice. You have two options:

```mojo
@staticmethod
fn index[C: ComparableCollectionElement](self: List[C], owned value: C, start: Int = 0) raises -> Int:
    return Self.index(self, value^, start, len(self))
```
this will call the second overload where the full implementation is. Or you have a second option, implementing only one signature:

```mojo
@staticmethod
fn index[C: ComparableCollectionElement](self: List[C], owned value: C, start: Int = 0, end: Optional[Int] = None) raises -> Int:
    if end is None:
        end = len(self)
    ...
```

Reply:
Thanks for bringing this up. I was also looking forward to implementing this, but I first wanted to have an acceptable version of `index`. Now that it's approved, I also going to merge those methods two by the help of `optional` in Mojo.
This also means we can pass `end` parameter without the need for specifying `start`. So I also added some new tests to cover this scenario.